src/core/create_compilers/RollupCompiler.ts

```diff
@@ -154,11 +158,40 @@ export default class RollupCompiler {
 
 		const that = this;
 
-		// TODO this is hacky. refactor out into an external rollup plugin
-		(mod.plugins || (mod.plugins = [])).push(css_chunks({ injectImports: true }));
-		if (!/[\\/]client\./.test(entry_point)) {
-			return mod;
-		}
+		/**
+		 * Track css code hashes for proper chunk hashing
+		 */
+		const css_hashing: Plugin = {
+			name: 'sapper-css-hashing',
+			augmentChunkHash(this: PluginContext, chunk: PreRenderedChunk) {
+				let hash: Hash;
+
+				const module_ids = Object.keys(chunk.modules).map(module_id => ({ module_id, module_path: path.parse(module_id) }));
+				module_ids.forEach(({ module_id, module_path }) => {
```

Comment:
I think this logic might be clearer if you instead did a `filter` to find all the matching `module_id`s. Then, after that, if there are any matching IDs you can create a hash from those IDs

Reply:
I know this current trend to use chained es6 array functions. But to be honest, I never got why I should optimise script language iteration code for "readability" or "easy to understand". This is not an environment where a compiler will optimise this kind of unnecessary multi iteration for you. If you ask me I would only "allow" es6 array functions if they are really just a short variant of a "simple" iteration. I always encourage other devs I work with to first think of a solution with as few iterations as possible and, if needed, use additional variables to save or determine necessary states etc.
I have seen whole app code foundations becoming pretty laggy just because the extensive use of chained array functions, lately.
I can change it, but I wouldn't suggest it tbh (and I don't really want to participate coding this unnecessary multi iterate way, no matter how many data gets processed 😛)

Reply:
I didn't mean to suggest you have to use chained es6 array functions and don't really care if you want to use `for` loops or something else. My main thought here was more about the structure of the operations. It took me awhile to understand what exactly is happening. I think if it were separated into two steps of first collecting the relevant IDs and then second building the hash from those IDs it'd be a bit easier to understand compared to building the hash as we go.

Reply:
My main point was to prefer a single iteration approach via variables (in this case `hash`) over multi iteration. I will change it, but I‘m still not a fan of using additional unnecessary iterations to make it "easier to understand".

Reply:
thanks. I think the number of elements being iterated over here will be quite limited, so there should not be any measurable performance impact, which is why I prefer the readability approach

Reply:
Ok, done 😬